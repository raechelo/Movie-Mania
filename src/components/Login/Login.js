import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';

export class Login extends Component {
  constructor() {
    super()
    this.state = {
      email: '',
      password: '',
      status: 0
    }
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.userLogin(this.state);
  }

  userLogin = (state) => {
    const {email, password} = state
    const url = 'http://localhost:3000/api/users'
    const options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        email,
        password
      })
    }
    fetch(url, options)
    .then(response => {
      this.setState({status: response.status})
      console.log(response.ok)
      response.ok ? this.loginUser() : this.showError();
    })
    .catch(error => console.log('error', error));
  }

  loginUser = () => {
    // Need to make redux action to login user
  }

  showError = () => {
    const errorText = document.querySelector('.error');
    errorText.style.color = '#760a0a';
  }

  handleChange = (e) => {
    const {name, value} = e.target;
    this.setState({[name]: value});
  }

  render() {
    return (
      <form className="user-form" onSubmit={this.handleSubmit}>
        <h2>Login</h2>
        <p class="error">Email and Password do not match</p>
        <label htmlFor="email">Email</label>
        <input onChange={ this.handleChange } name="email" type="email" id="email" placeholder="Email" />
        <label htmlFor="password">Password</label>
        <input onChange={ this.handleChange } name="password" type="password" id="password" placeholder="Password" />
        <button>Login</button>
        {this.state.status === 200 && <Redirect to="/" />}
      </form>
    )
  }
}

export default Login;