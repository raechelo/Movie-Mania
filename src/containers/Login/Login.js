import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { logInUser } from '../../actions';
import { fetchData } from '../../utils/api/fetchData';
 
export class Login extends Component {
  constructor() {
    super()
    this.state = {
      email: '',
      password: '',
      status: ''
    }
  }

  handleSubmit = (e) => {
    e.preventDefault();
    this.userLogin(this.state);
  }

  userLogin = (state) => {
    const {email, password} = state;
    const url = 'http://localhost:3000/api/users'
    const options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        email,
        password
      })
    }
    fetchData(url, options)
    .then(result => {
      this.setState({status: result.status});
      this.props.logInUser(result.data);
    })
    .catch(error => this.showError());
  }

  showError = () => {
    const errorText = document.querySelector('.error');
    errorText.style.color = '#760a0a';
  }

  handleChange = (e) => {
    const {name, value} = e.target;
    this.setState({[name]: value});
  }

  render() {
    return (
      <form className="user-form" onSubmit={this.handleSubmit}>
        <h2>Login</h2>
        <p className="error">Email and Password do not match</p>
        <label htmlFor="email">Email</label>
        <input onChange={ this.handleChange } name="email" type="email" id="email" placeholder="Email" />
        <label htmlFor="password">Password</label>
        <input onChange={ this.handleChange } name="password" type="password" id="password" placeholder="Password" />
        <button>Login</button>
        {this.state.status && <Redirect to="/" />}
      </form>
    )
  }
}

Login.propTypes = {
  logInUser: PropTypes.func
}


export const mapDispatchToProps = (dispatch) => ({
  logInUser: (user) => dispatch(logInUser(user))
})

export default connect(null, mapDispatchToProps)(Login);