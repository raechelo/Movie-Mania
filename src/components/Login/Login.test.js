import React from 'react';
import { shallow } from 'enzyme';
import { Login } from './Login';

describe('Login', () => {

  let wrapper;
  let mockUserLogin = jest.fn()

  beforeEach(() => {
    wrapper = shallow( <Login /> )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  //this test will probably need to be changed once we get the thunk in place for userLogin
  it.skip('should handle submit', () => {
    const e = { target: { className: 'user-form' },preventDefault: () => {} }
    const expected = {email: 'godofolympus@gmail.com', password: 'GoodBoye', status: 200}

    wrapper.instance().handleSubmit(e);

    expect(mockUserLogin).toHaveBeenCalledWith(...expected);
  });

  it('should handle change', () => {
    wrapper.setState({ email: '', password: ''})
    const e = { target: { name: 'email', value: 'godofolympus@gmail.com' } }

    wrapper.instance().handleChange(e);
    expect(wrapper.state('email')).toEqual('godofolympus@gmail.com')
  });
});