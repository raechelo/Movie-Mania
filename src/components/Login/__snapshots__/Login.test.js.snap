// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot with all the data passed in 1`] = `
<form
  className="user-form"
  onSubmit={[Function]}
>
  <h2>
    Login
  </h2>
  <p
    className="error"
  >
    Email and Password do not match
  </p>
  <label
    htmlFor="email"
  >
    Email
  </label>
  <input
    id="email"
    name="email"
    onChange={[Function]}
    placeholder="Email"
    type="email"
  />
  <label
    htmlFor="password"
  >
    Password
  </label>
  <input
    id="password"
    name="password"
    onChange={[Function]}
    placeholder="Password"
    type="password"
  />
  <button>
    Login 
    <i
      class="fas fa-angle-right"
    />
  </button>
</form>
`;
